packages/@aws-cdk/aws-ecs-patterns/lib/base/application-multiple-target-groups-service-base.ts

```diff
@@ -0,0 +1,612 @@
+    return new AwsLogDriver({ streamPrefix: prefix });
+  }
+
+  protected findListener(name?: string): ApplicationListener {
+    if (!name) {
+      return this.listener;
+    }
+    for (const listener of this.listeners) {
+      if (listener.node.id === name) {
+        return listener;
+      }
+    }
+    throw new Error(`Listener ${name} is not defined. Did you define listener with name ${name}?`);
+  }
+
+  protected registerECSTargets(service: BaseService, container: ContainerDefinition, targets: ApplicationTargetProps[]): ApplicationTargetGroup {
+    for (const targetProps of targets) {
+      const targetGroup = this.findListener(targetProps.listener).addTargets(`ECSTargetGroup${container.containerName}${targetProps.containerPort}`, {
+        port: 80,
+        targets: [
+          service.loadBalancerTarget({
+            containerName: container.containerName,
+            containerPort: targetProps.containerPort,
+            protocol: targetProps.protocol
+          })
+        ],
+        hostHeader: targetProps.hostHeader,
+        pathPattern: targetProps.pathPattern,
+        priority: targetProps.priority
+      });
+      this.targetGroups.push(targetGroup);
+    }
+    if (this.targetGroups.length === 0) {
+      throw new Error('At least one target group should be specified.');
+    }
+    return this.targetGroups[0];
+  }
+
+  protected addPortMappingForTargets(container: ContainerDefinition, targets: ApplicationTargetProps[]) {
+    for (const target of targets) {
+      if (!container.findPortMapping(target.containerPort, target.protocol || Protocol.TCP)) {
+        container.addPortMappings({
+          containerPort: target.containerPort,
+          protocol: target.protocol
+        });
+      }
+    }
+  }
+
+  /**
+   * Create log driver if logging is enabled.
+   */
+  private createLogDriver(enableLoggingProp?: boolean, logDriverProp?: LogDriver): LogDriver | undefined {
+    const enableLogging = enableLoggingProp !== undefined ? enableLoggingProp : true;
+    const logDriver = logDriverProp !== undefined
+                        ? logDriverProp : enableLogging
+                          ? this.createAWSLogDriver(this.node.id) : undefined;
+    return logDriver;
+  }
+
+  private configListener(protocol: ApplicationProtocol, props: ListenerConfig): ApplicationListener {
+    const listener = this.createListener(props.listenerName, props.loadBalancer, protocol, props.port);
+    let certificate;
+    if (protocol === ApplicationProtocol.HTTPS) {
+      certificate = this.createListenerCertificate(props.listenerName, props.certificate, props.domainName, props.domainZone);
+    } else {
+      certificate = undefined;
+    }
+    if (certificate !== undefined) {
+      listener.addCertificateArns(`Arns${props.listenerName}`, [certificate.certificateArn]);
+    }
+
+    return listener;
+  }
+
+  private validateInput(props: ApplicationMultipleTargetGroupsServiceBaseProps) {
```

Comment:
Does it make sense to do validation on props.loadbalancers and their listeners here too?

Reply:
:+1: